### PR TITLE
Use PicoCLI over custom args parsing

### PR DESCRIPTION
**Pull request by Brutus5000**
Due to the fact that the cli launch process is decoupled from start function, it is now also possible to start the ICE adapter from inside another Java process.

(I didn't test it, maybe JavaFX fails if to main windows are launched)

### CODE REVIEW

**Comment by Brutus5000** [comment]
I wouldn't recommend launching it from inside anything else yet due to the usage of System.exit in many places

File: ice-adapter/src/main/java/com/faforever/iceadapter/CliLauncher.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import picocli.CommandLine;
+
+import java.util.concurrent.Callable;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+public class CliLauncher implements IceOptions, Callable<Integer> {
```
Comment (Sheikah45) [inline ice-adapter/src/main/java/com/faforever/iceadapter/CliLauncher.java:13]: This needs an `@Command` annotation to work. I would also suggest adding the mixinStandardHelpOptions=true as well as a version provider

**Review by Sheikah45 — CHANGES_REQUESTED** [review]
I am also not sure if picocli picks up the anntations through the interface. I know normally it works through composition, but will test it out.

**Comment by Sheikah45** [comment]
Yeah just tested it, picocli doesn't work through inheritance. So either need to make the iceOptions a filed with the @ ArgGroup annotation or put the annotations directly in CLILauncher

File: ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java [hunk 2]
```diff
@@ -38,11 +36,24 @@ public class IceAdapter {
...
     public static volatile GameSession gameSession;
 
-    public static void main(String args[]) {
+    public static void main(String[] args) {
+        final CommandLine commandLine = new CommandLine(IceOptions.class);
+
+        try {
+            commandLine.parseArgs(args);
+        } catch (CommandLine.PicocliException e) {
+            System.out.println(e.getMessage());
```
Comment (Sheikah45) [inline ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java:45]: Don't we have a logger for this XD

**Review by Sheikah45 — APPROVED** [review]

**Pull request merged**